source/loader/layers/sanitizer/asan_interceptor.cpp

```diff
@@ -0,0 +1,617 @@
+//===----------------------------------------------------------------------===//
+/*
+ *
+ * Copyright (C) 2023 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * @file ur_sanitizer_layer.cpp
+ *
+ */
+
+#include "asan_interceptor.hpp"
+#include "device_sanitizer_report.hpp"
+#include "ur_sanitizer_layer.hpp"
+
+namespace ur_sanitizer_layer {
+
+namespace {
+
+// These magic values are written to shadow for better error
+// reporting.
+const int kUsmDeviceRedzoneMagic = (char)0x81;
```

Comment:
cast isn't needed.
```suggestion
constexpr int kUsmDeviceRedzoneMagic = 0x81;
```

Reply:
I need the sign bit expansion here, so the cast is needed.

Reply:
You can still use constexpr

Reply:
done